emctl/cmd/transformer/generator/visitor.go

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright (c) 2017, MegaEase
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package generator
+
+import (
+	"fmt"
+	"go/ast"
+	"strings"
+
+	"github.com/dave/jennifer/jen"
+	"github.com/pkg/errors"
+)
+
+type (
+	meshClientVisitor struct {
+		err          error
+		getterID     string
+		resourceType ResourceType
+		builder      interfaceMethodBuilder
+	}
+
+	converter struct {
+		err      error
+		funcType *ast.FuncType
+		funcName string
+		imports  []*ast.ImportSpec
+	}
+)
+
+// NewVisitor create an InterfaceVisitor object to generate code while traveling the interface.
+func NewVisitor(resourceType ResourceType) InterfaceVisitor {
+	return &meshClientVisitor{resourceType: resourceType, builder: &interfaceBuilder{}}
+}
+
+func (m *meshClientVisitor) visitorBegin(imports []*ast.ImportSpec, spec *InterfaceFileSpec) error {
+	spec.Buf.PackageComment(`Copyright (c) 2017, MegaEase
```

Comment:
```
/*
 * Copyright (c) 2017, MegaEase
 * All rights reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
```

This might eliminate the warning.

Reply:
The warning message can only be eliminated by inserting an empty line between the package comment (license declaration) and the `package XXXX` of the statement. But Jennifer does not support doing this...

Reply:
Okay, not a big problem.